lib/src/Uncertainty/Algorithm/OrthogonalBasis/OrthogonalProductPolynomialFactory.cxx

```diff
@@ -154,6 +154,70 @@ String OrthogonalProductPolynomialFactory::__repr__() const
          << " measure=" << measure_;
 }
 
+String OrthogonalProductPolynomialFactory::__str__(const String & /*offset*/) const
+{
+  return __repr_markdown__();
+}
+
+/* String formatter with blank space right padding */
+String OrthogonalProductPolynomialFactory::padString(const String & string, const UnsignedInteger columnWidth) const
+{
+  const String intermediateString = OSS() << string;
+  const UnsignedInteger rightPadding = intermediateString.size() < columnWidth ? columnWidth - intermediateString.size() : 0;
+  OSS oss;
+  oss << intermediateString << String(rightPadding, ' ');
+  return oss;
+}
+
+/* Get the input marginal polynomial class name */
+String OrthogonalProductPolynomialFactory::getMarginalPolynomialName(const UnsignedInteger marginalIndex) const
+{
+  String className;
+  const OrthogonalUniVariatePolynomialFamily univariatePolynomial = coll_[marginalIndex];
+  const String basicClassName = univariatePolynomial.getImplementation()->getClassName();
+  if (basicClassName == "StandardDistributionPolynomialFactory")
+  {
+    const StandardDistributionPolynomialFactory* p_factory = dynamic_cast<const StandardDistributionPolynomialFactory*>(univariatePolynomial.getImplementation().get());
+    const Bool hasSpecificFamily = p_factory->getHasSpecificFamily();
```

Comment:
You must check that p_factory is not null. Otherwise use a const_cast.

Reply:
Ok. I have to do that.

Reply:
Fixed at b1fc38160. Used [ParametricGradient.cxx#LL74C5-L74C20](https://github.com/openturns/openturns/blob/457622ecf5264389f3739316e9c8cdc92aaea083/lib/src/Base/Func/ParametricGradient.cxx#LL74C5-L74C20) as a template. Please resolve this discussion if appropriate.